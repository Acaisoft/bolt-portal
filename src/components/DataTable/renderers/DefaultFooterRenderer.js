import React from 'react'
import { TableFooter, TableRow, TableCell } from '@material-ui/core'

function DefaultFooterRenderer({ classes, columns, data, isLoading }) {
  if (isLoading) {
    return null
  }

  return (
    <TableFooter>
      <TableRow className={classes.footer}>
        {columns.map(({ renderFooter, title, render, ...cellProps }) => (
          <TableCell {...cellProps}>
            {typeof renderFooter === 'function' ? renderFooter(data) : null}
          </TableCell>
        ))}
      </TableRow>
    </TableFooter>
  )
}

export default DefaultFooterRenderer
