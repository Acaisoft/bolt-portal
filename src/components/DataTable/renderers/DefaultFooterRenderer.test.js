import React from 'react'
import { shallow } from 'enzyme'

import { ClassesProxy } from '~utils/tests/mocks'

import DefaultFooterRenderer from './DefaultFooterRenderer'
import { children, columnSettings, data } from '../DataTable.test.mock'

const initBodyRenderer = overrides => {
  const mockProps = {
    classes: ClassesProxy,
    columns: columnSettings,
    data,
    ...overrides,
  }

  const wrapper = shallow(<DefaultFooterRenderer {...mockProps} />)
  return { mockProps, wrapper }
}

describe('component: DefaultFooterRenderer', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initBodyRenderer()
      expect(wrapper).toBeTruthy()
    })

    it('should render loader if isLoading is set', () => {
      const { wrapper } = initBodyRenderer({ children, isLoading: true })
      expect(wrapper).toMatchSnapshot()
    })

    it('should render footer cells according to column settings', () => {
      const { wrapper } = initBodyRenderer({ children })
      expect(
        wrapper
          .find('TableCellMock')
          .at(0)
          .text()
      ).toEqual('')
      expect(
        wrapper
          .find('TableCellMock')
          .at(1)
          .text()
      ).toEqual('Total age: 66')
    })
  })
})
