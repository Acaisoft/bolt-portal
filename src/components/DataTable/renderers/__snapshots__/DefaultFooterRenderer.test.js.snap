// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: DefaultFooterRenderer rendering should render loader if isLoading is set 1`] = `""`;
